internal/provider/module_resource.go

```diff
@@ -540,6 +591,30 @@ func toDependenciesFromModel(ctx context.Context, dependencies basetypes.MapValu
 	return result, nil
 }
 
+func toModuleParamsFromModel(ctx context.Context, moduleParams basetypes.MapValue) (map[string]canyoncp.ModuleParamItem, error) {
+	result := make(map[string]canyoncp.ModuleParamItem)
+	if !moduleParams.IsNull() && !moduleParams.IsUnknown() {
+		for key, value := range moduleParams.Elements() {
+			paramObj, ok := value.(basetypes.ObjectValue)
+			if !ok {
+				return nil, fmt.Errorf("expected object value for module param %s", key)
```

Comment:
```suggestion
				return nil, fmt.Errorf("expected object value for module param %q", key)
```
to escape `key` values when forming string

Reply:
@tnako thankfully because param keys can only be valid terraform identifiers, they don't contain any characters that need escaping. Only a-z, 0-9, -, and _